### ISSUE TITLE
[1.19] Reloading a saved game break biome placement

### ISSUE THREAD

**Issue by ValkyrieSama** [Glitchfiend/TerraBlender#90]
### What's the issue you encountered?

Hi!

Each time I exit and reenter a saved game, the biome arrangement switch around creating a grid like pattern on the divisions.

For example, when I explore a desert, I quit the game, then reload and now it's plains on the border I didn't yet explored. If I quit and reload again, it can be whatever else past the horizon of the plains.

![image](https://user-images.githubusercontent.com/11657341/191300581-5f95277f-e7a7-4e15-9d74-cd9698d28912.png)

### How can the issue be reproduced?

Just install TerraBlender with forge 43.1.25. 

You can use Terralith or Oh the biome you'll go but it doesn't change anything. It's observable in vanilla terrain gen too.

Just create a new save, get in the game, move around a bit then quit. Reload, move past the explored chunks then observe the biome change.

### Mod Version

2.0.1.127

### Additional information

I changed the rendering distance to 10 chunks, not sure if it change anything though. 
I also create the worlds with large biomes enabled.

**Comment by Adubbz** [comment]
This usually occurs when your world's seed gets reset for whatever reason. Please reproduce with ONLY TerraBlender and BYG installed. Logs would also be helpful.

**Comment by ValkyrieSama** [comment]
In this one, I used, BYG, Terralith, Terrablender and JourneyMap so I could easily see the split:

![2022-09-20_20 18 42](https://user-images.githubusercontent.com/11657341/191387107-ef6e0240-ac08-4156-a2ef-f30aa3a07051.png)

![image](https://user-images.githubusercontent.com/11657341/191387280-5a78d90e-822b-430b-a286-3f4a155aba26.png)

and here's the log my kind sir.

[2022-09-20-2.log](https://github.com/Glitchfiend/TerraBlender/files/9611992/2022-09-20-2.log)

With one with BYG and TerraBlender, I can do the exact same thing.
![2022-09-20_20 29 03](https://user-images.githubusercontent.com/11657341/191388063-b22a6406-ece9-4133-9812-3ef0f77d3173.png)

**Comment by Abalieno** [comment]
I've researched this extensively and as far as I know it's Terralith.

The only working solution I could find was to load Terralith (and Incendium or Nullscape, same thing) not as a mod, but as a datapack with a fixed seed: https://seedfix.stardustlabs.net/

There's a mod titled "Paxi" that can load data packs and resource packs automatically. So i used that.

Sometimes the seed "hopping" moves to other dimensions. But fixing the seed SEEMS to have stopped this problem. I think there's something that goes crazy if the seed in the mod or datapack is set to zero.

Btw, I see that problem is about 1.19 and Forge, whereas I'm reporting about 1.18 and Fabric. The problem looks about the same, and probably the solution works.

**Comment by catter1** [comment]
> I've researched this extensively and as far as I know it's Terralith.
> 
> The only working solution I could find was to load Terralith (and Incendium or Nullscape, same thing) not as a mod, but as a datapack with a fixed seed: https://seedfix.stardustlabs.net/
> 
> There's a mod titled "Paxi" that can load data packs and resource packs automatically. So i used that.
> 
> Sometimes the seed "hopping" moves to other dimensions. But fixing the seed SEEMS to have stopped this problem. I think there's something that goes crazy if the seed in the mod or datapack is set to zero.
> 
> Btw, I see that problem is about 1.19 and Forge, whereas I'm reporting about 1.18 and Fabric. The problem looks about the same, and probably the solution works.

Hi, I'm from Terralith. I'm trying to look into the problem, and curious on what your discoveries were (and exactly which you tried). Which encounter the bug, which don't, and which have you not yet tried?
- Fabric 1.19.x
- Forge 1.19.x
- Fabric 1.18.2
- Forge 1.18.2

Thanks!

**Comment by ValkyrieSama** [comment]
Yea my bad, seems like once a world has been created with terralith, on any reload, the seed is reset. I'm not certain why. In the end it's not a TerraBlender issue.

**Comment by ValkyrieSama** [comment]
But it's happening on Forge and Fabric 1.19.2 when I tested it. The seed fix from Terralith doesn't solve the issue in 1.19 but it does in 1.18

**Comment by Abalieno** [comment]
A bit of an update after tons of testing.

The 1.19 forge issue seems to be a different problem due to forge.

The 1.18 fabric issue seems to NOT be realted to Terralith, or biomes, but seed corruption due to Minecraft bug when some mods try to change the seed of a dimension from the standard /seed one.

Incendium tries to set the Nether seed to 41, and on reload seed corruption spreads to other dimensions randomly.

Same for Ad Astra, that creates dimensions for other planets and that causes significant seed corruption spreading to other dimensions, like Overworld, or the dimension that the mod Spectrum adds.

Basically, as long all dimensions have the same seeds, no problem. If a mod tries to set a separate seed, on a full reload of the modpack the seeds from other dimensions start getting reset.

There are only two solutions:
- Have ALL dimensions share the same general seed.
- Have ALL dimensions have a different but fixed specific seed.

If there's a dimension that sets a specific seed, and then a mod adds another dimension with variable seed, then that dimension will have seed corruption on reload.
